### PR TITLE
CLDR-17399 Migrate from deprecated boxed primitive constructors

### PR DESCRIPTION
**Pull request by mihnita**
CLDR-17399

- [x] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Comment by macchiati** [comment]
Were these changes generated by a refactoring tool or ide?

**Review by srl295 — DISMISSED** [review]

**Comment by srl295** [comment]
org.json is external but has already diverged

**Comment by srl295** [comment]
needs mvn spotless:apply

**Comment by mihnita** [comment]
> Were these changes generated by a refactoring tool or ide?

I wish  :-)

Detected by erroprone. I added a profile for maven so that we can run from command line.
(not a friendly report though)

I use the report as a guideline, but did the fixes in IntelliJ, with assistance.
Could have accepted the fixes "as is", but they were not always the best option.

For example:
```
Integer foo = new Integer(123);
```
Proposed fix:
```
Integer foo = Integer.valueOf(123);
```

But that only introduced another errorprone issue (not reported if you limit yourself to test for a one issue type only).
What I did in the end was:
```
Integer foo = 123;
```

TLDR: IDE assisted

**Comment by mihnita** [comment]
> org.json is external but has already diverged
I will reconcile.

> needs mvn spotless:apply
Done. I didn't know about it, I don't touch CLDR too much (at all :-).

Thanks,
M

**Review by markusicu — APPROVED** [review]
lgtm
I don't know CLDR policy on org.json code. I am surprised that CLDR has "vendored" code in its repo, I thought such code was to be replaced by Maven dependencies.
Steven seems to indicate that it's ok to change this code anyway.

File: tools/cldr-apps/src/main/java/org/json/JSONArray.java [hunk 1]
```diff
@@ -555,7 +555,7 @@ public JSONArray put(double value) throws JSONException {
      * @return this.
      */
     public JSONArray put(int value) {
-        put(new Integer(value));
+        put(value);
```
Comment (srl295) [inline tools/cldr-apps/src/main/java/org/json/JSONArray.java:558]: @mihnita BTW, postscript, this is a stack overflow (recursive)

File: tools/cldr-apps/src/main/java/org/json/JSONArray.java [hunk 2]
```diff
@@ -566,7 +566,7 @@ public JSONArray put(int value) {
      * @return this.
      */
     public JSONArray put(long value) {
-        put(new Long(value));
+        put(value);
```
Comment (srl295) [inline tools/cldr-apps/src/main/java/org/json/JSONArray.java:569]: @mihnita BTW, postscript, this is a stack overflow (recursive)
